integration-tests/slack/notify-circleci-test-results.js

```diff
@@ -0,0 +1,77 @@
+const _ = require('lodash/fp')
+const { getFailedTestNamesFromArtifacts } = require('./circleci-utils')
+const { getMessageBlockTemplate } = require('./message-templates')
+const { postMessage } = require('./post-message')
+const testsInfo = require('./slack-notify-channels.json')
+
+
+/**
+ * Get array of Slack channel IDs for succeeded job notification
+ * @returns {Array[string]}
+ */
+const getAllSlackChannelsForPassedJob = () => {
+  return _.map(_.get('id'), testsInfo.pass)
+}
+
+/**
+ * Get all Slack channel IDs and test names for failed job notification
+ * @returns {Map<string, Array[string]>}
+ */
+const getAllSlackChannelsForFailedJob = () => {
+  const allChannelsAndTests = new Map()
+  _.forEach(item => {
+    const channelId = _.get('id', item)
+    const testNames = _.map(test => test.name, item.tests)
+    allChannelsAndTests.set(channelId, testNames)
+  }, testsInfo.fail)
+  return allChannelsAndTests
+}
+
+/**
+ * Get a map object of Slack channel IDs and array of test names for failed tests notification
+ * @param {Array[string]} failedTests
+ * @returns {Map<string, Array[string]>} A map object where key is channel_id, value is test_names array
+ */
+const getFailedTestsAndChannelIDs = failedTests => {
+  const allChannelsAndTests = getAllSlackChannelsForFailedJob()
+
+  const filteredIncludeOnlyFailedTests = new Map() // Map<string, Array[string]>
+  _.forEach(test => {
+    const channelIdsForFailedTest = Array.from(allChannelsAndTests.keys())
+      .filter(key => allChannelsAndTests.get(key).includes(test) || allChannelsAndTests.get(key).length === 0) // A channel without a list of tests means notify that channel for any failing test
+    _.forEach(channelId => {
+      if (!filteredIncludeOnlyFailedTests.has(channelId)) {
+        filteredIncludeOnlyFailedTests.set(channelId, [])
+      }
+      filteredIncludeOnlyFailedTests.get(channelId).push(test)
+    }, channelIdsForFailedTest)
+  }, failedTests)
+
+  return filteredIncludeOnlyFailedTests
+}
+
+// Post CircleCI UI test report to Slack channels
+const notifyCircleCITestResults = async () => {
+  const failedTestNames = await getFailedTestNamesFromArtifacts()
+
+  if (failedTestNames.length === 0) {
+    // Slack notification: CircleCI job succeeded
+    const channelIDs = getAllSlackChannelsForPassedJob()
+    const messageBlocks = getMessageBlockTemplate(failedTestNames)
+    // Using _.forEach to post same message to each Slack channels separately because a Slack issue: No way to post the same message to multiple channels at once.
+    // See: https://github.com/slackapi/bolt-js/issues/696
+    _.forEach(async channelId => await postMessage({ channel: channelId, blocks: messageBlocks }), channelIDs)
```

Comment:
Here (and below) is a perfect example to support my other comment. Here `_.forEach` is appropriate[*] because you want a set of side-effects for each item in the list.

[*] One could argue that `await Promise.all(_.map(async channelId => ...))` would be better because the order of execution of the side-effects doesn't matter.